test/scripts/e2e_subs/e2e-app-simple.sh

```diff
@@ -24,15 +24,17 @@ printf 'int 1' > "${TEMPDIR}/simplev1.teal"
 # Fail in creating app with v1 approval program
 RES=$(${gcmd} app create --creator ${ACCOUNT} --approval-prog "${TEMPDIR}/simplev1.teal" --clear-prog "${TEMPDIR}/simple.teal" --global-byteslices 0 --global-ints ${GLOBAL_INTS} --local-byteslices 0 --local-ints 0 2>&1 || true)
 EXPERROR='program version must be >= 2 for this transaction group'
-if [[ $RES != *"${EXPERROR}"* ]]; then
+EXPERRORUNICODE='program version must be \u003e= 2 for this transaction group'
```

Comment:
v2 API returns a unicode-escaped character for the `>` sign - changed the tests to accept both.

Reply:
@algochoi For my understanding:  Did the test change rather than disabling `SetEscapeHTML` because you feel it's the more appropriate change or to avoid changing a public response value?

If the latter, I'm wondering if it's worth raising for a broader consideration.  At least for goal usage, it feels strange to me to see escaped values.

Reply:
>  to avoid changing a public response value

Yes, I think changing a public response is outside the scope of this PR. I agree that for goal, it's strange to see this escaped, but it looks like we'd have to change the behavior of the server error handler, which would affect more than goal usage.

Reply:
Seeking additional opinions - Can other reviewers chime in on the thread?

* It's unclear to me that v2 APIs _intentionally_ default to ` SetEscapeHTML(true) `.  
  * If it _is_ intentional, I'd be more convinced the PR presents desired behavior.
  * Otherwise, I think _at least_ goal output with escaped characters is undesired behavior.
* Within the time I had available to search for analogous issues, I see asset name, unit name, and URL are also encoded as `string` + liable to have the same concern.  Here's a relevant snippets: 
https://github.com/algorand/go-algorand/blob/f37bb6ef9d9eb14b78699f2af48738d07c2611c1/daemon/algod/api/server/v2/account.go#L450-L463
* During my limited search, I did _not_ see a test illustrating the escaped behavior to help confirm the concern applies to assets.

Reply:
> Otherwise, I think at least goal output with escaped characters is undesired behavior.

I agree with this - we can decode the output on the goal client at the minimum.

Reply:
I took a closer look and the `echo` v4.1.17 dependency that we use does not allow custom flags (`SetEscapeHTML`) to be set in the encoder: 
https://github.com/labstack/echo/blob/cb84205219f6b0962ab5bb39ab6c0911f5f1652d/context.go#L456

@winder @jasonpaulos : any thoughts on whether we should escape these characters on the server/client side or leave as is?

Reply:
I don't think we set the escape characters intentionally. @algochoi found the echo code used by the v2 API, there is a convenience function which automatically encodes things using their own encoder.

On the other hand in `codec.go` we set `HSONHandle.HTMLCharsAsIs = true`, which seems to tell the encoder to use an escaped format.

Reply:
I'm fine with a decode pass in goal

Reply:
I think the server behavior should remain the same. I was surprised to learn that the reason the escaped characters still appear in goal is because the error response is never decoded.

I opened https://github.com/algochoi/go-algorand/pull/27 as an attempt to address this, let me know what you think.

Reply:
Thanks for the replies - I think I still have questions though I can address it out-of-band to the PR.

Provided https://github.com/algochoi/go-algorand/pull/27 is merged in, I'm good to resolve the thread.

Reply:
Thanks @jasonpaulos ! I merged the PR in.